consumer/consumererror/internal/statusconversion/conversion.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package statusconversion // import "go.opentelemetry.io/collector/consumer/consumererror/internal/statusconversion"
+
+import (
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func GetHTTPStatusCodeFromStatus(s *status.Status) int {
+	// See https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures
+	// to see if a code is retryable.
+	// See https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures-1
+	// to see a list of retryable http status codes.
+	switch s.Code() {
+	// Retryable
+	case codes.Canceled, codes.DeadlineExceeded, codes.Aborted, codes.OutOfRange, codes.Unavailable, codes.DataLoss:
+		return http.StatusServiceUnavailable
+	// Retryable
+	case codes.ResourceExhausted:
+		return http.StatusTooManyRequests
+	// Not Retryable
```

Comment:
Why 500 isn't retriable?

Reply:
Retry eligibility is taken solely from the OTLP specification, which only lists four error codes as retryable: https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures-1